File: scripts/parse-or-catalog.js
const fs = require("fs");
const path = require("path");

const TEMP = process.env.TEMP;

function parseCsv(text) {
  const rows = [];
  for (const line of text.split(/\r?\n/)) {
    const cols = [];
    let cur = "";
    let q = false;
    for (let i = 0; i < line.length; i++) {
      const c = line[i];
      if (c === '"') {
        q = !q;
        continue;
      }
      if (c === "," && !q) {
        cols.push(cur);
        cur = "";
        continue;
      }
      cur += c;
    }
    cols.push(cur);
    rows.push(cols);
  }
  return rows;
}

function parse(file, metal) {
  const text = fs.readFileSync(path.join(TEMP, file), "utf8");
  const rows = parseCsv(text);
  const groups = [
    [1, 2, 3, "ARMES"],
    [5, 6, 7, "ACCESSOIRES_ARMES"],
    [9, 10, 11, "ACCESSOIRES"],
    [13, 14, 15, "DROGUES"],
  ];
  const items = [];
  for (const row of rows.slice(3)) {
    for (const [pc, nc, ic, cat] of groups) {
      if (row.length <= Math.max(pc, nc, ic)) continue;
      const price = (row[pc] || "").trim().replace(/ /g, "");
      const name = (row[nc] || "").trim();
      const wid = (row[ic] || "").trim();
      if (!name || !price) continue;
      const digits = price.replace(/[^\d]/g, "");
      if (!digits) continue;
      const p = Number(digits);
      if (p <= 0) continue;
      const slug = (wid || name)
        .toLowerCase()
        .replace(/[^a-z0-9_]+/g, "_");
      items.push({
        id: `${metal}-${slug}`,
        name,
        wikiId: wid || null,
        category: cat,
        price: p,
        metal,
      });
    }
  }
  const seen = new Set();
  const out = [];
  for (const it of items) {
    const k = `${it.name.toUpperCase()}|${it.price}|${it.category}`;
    if (seen.has(k)) continue;
    seen.add(k);
    out.push(it);
  }
  return out;
}

const rouge = parse("or-rouge.csv", "rouge");
const bleu = parse("or-bleu.csv", "bleu");
const out = "C:/Users/Shadow/cveshop/src/lib/or-market-catalog-data.json";
fs.writeFileSync(out, JSON.stringify({ rouge, bleu }, null, 2));
console.log("rouge", rouge.length, "bleu", bleu.length);

File: scripts/parse-or-catalog.py
import csv
import json
import os
import re
from pathlib import Path

TEMP = os.environ["TEMP"]


def parse(path: str, metal: str):
    text = Path(path).read_text(encoding="utf-8", errors="replace")
    rows = list(csv.reader(text.splitlines()))
    items = []
    groups = [
        (1, 2, 3, "ARMES"),
        (5, 6, 7, "ACCESSOIRES_ARMES"),
        (9, 10, 11, "ACCESSOIRES"),
        (13, 14, 15, "DROGUES"),
    ]
    for row in rows[3:]:
        if not row:
            continue
        for pc, nc, ic, cat in groups:
            if len(row) <= max(pc, nc, ic):
                continue
            price = (row[pc] or "").strip().replace(" ", "")
            name = (row[nc] or "").strip()
            wid = (row[ic] or "").strip()
            if not name or not price:
                continue
            digits = re.sub(r"[^\d]", "", price)
            if not digits:
                continue
            p = int(digits)
            if p <= 0:
                continue
            slug = re.sub(r"[^a-z0-9_]+", "_", (wid or name).lower())
            items.append(
                {
                    "id": f"{metal}-{slug}",
                    "name": name,
                    "wikiId": wid or None,
                    "category": cat,
                    "price": p,
                    "metal": metal,
                }
            )
    seen = set()
    out = []
    for it in items:
        key = (it["name"].upper(), it["price"], it["category"])
        if key in seen:
            continue
        seen.add(key)
        out.append(it)
    return out


rouge = parse(os.path.join(TEMP, "or-rouge.csv"), "rouge")
bleu = parse(os.path.join(TEMP, "or-bleu.csv"), "bleu")
out = Path(r"C:\Users\Shadow\cveshop\src\lib\or-market-catalog-data.json")
out.write_text(
    json.dumps({"rouge": rouge, "bleu": bleu}, ensure_ascii=False, indent=2),
    encoding="utf-8",
)
print("rouge", len(rouge), "bleu", len(bleu), "->", out)
